### PR TITLE
Implement functionality for a judge to assign a case to an attorney

### LINKED ISSUE #4897: Allow a judge to assign an appeal to an attorney with POST /tasks
Related to https://github.com/department-of-veterans-affairs/caseflow/issues/4853

Judges need to be able to assign appeals to attorneys.

POST /tasks

```
{
  appeal_id: 48323209C
  appeal_type: "Legacy"
  attorney_id: 4
}
```

## AC
- Only judges can assign appeals to attorneys, if the user who POSTs to `/tasks` is not a judge, error (we'll have to change this later)
- Create a row in DECASS:
  - `defolder` should be the appeal id
  - `deatty` should be the attorney id
  - `deteam` should be the attorney/judge team id
  - `depdiff`—?
  - `deadusr` should be the judge’s slogid
  - `deadtim` should be the date (not hour/m/second) that the assignment is created
  - We can likely skip `dedeadline`
  - `deicr`—?
- Update the appeal's location to the attorney's location

### PR DESCRIPTION
**Pull request by aroltsch**
Resolves #4897 

### Description
Implement an endpoint for a judge to assign a case to an attorney
TODO: tackle depdiff and dedeadline at a later PR

### Testing Plan
1. Use Postman

```
POST /queue/tasks
params = {
  "appeal_id": 1,
  "appeal_type": "Legacy",
  "attorney_id": 1
}
```

2. Use command line

```
CaseAssignment.assign!(appeal_type: "Legacy", assigned_by: <User object>, assigned_to: <User object>, vacols_id: 2700904)
```

### CODE REVIEW

File: app/models/attorney_case_reviews/attorney_case_review.rb [hunk 1]
```diff
@@ -63,18 +61,9 @@ class << self
...
-          record.reassign_case_to_judge_in_vacols!
-          record.update_issue_dispositions! if record.type == "DraftDecision"
-        # :nocov:
-        rescue *EXCEPTIONS => e
-          Raven.capture_exception(e)
-          Rails.logger.warn(e)
-          raise ActiveRecord::Rollback
-        end
-        # :nocov:
+        record = create!(params)
```
Comment (aroltsch) [inline app/models/attorney_case_reviews/attorney_case_review.rb]: Alex, could you also validate this please!

File: lib/fakes/queue_repository.rb [hunk 2]
```diff
@@ -49,4 +49,8 @@ def self.appeals_from_tasks(_tasks)
   def self.reassign_case_to_judge!(_decass_hash)
     true
   end
+
+  def self.assign_case_to_attorney!(args)
```
Comment (aroltsch) [inline lib/fakes/queue_repository.rb]: I can work on removing fakes for Queue so we can get rid of this mess.

File: app/models/tasks/task_assignment.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class TaskAssignment
```
Comment (aroltsch) [inline app/models/tasks/task_assignment.rb]: What are your thoughts on this name?

File: app/controllers/queue_controller.rb [hunk 4]
```diff
@@ -1,9 +1,17 @@
 class QueueController < ApplicationController
   before_action :react_routed, :check_queue_out_of_service
-  before_action :verify_welcome_gate_access, except: :complete
+  before_action :verify_welcome_gate_access, except: [:complete, :create]
```
[the remark this answers is not in the document]
Comment (aroltsch) [reply app/controllers/queue_controller.rb]: Very true!

File: app/models/user.rb [hunk 5]
```diff
@@ -42,6 +42,14 @@ def vacols_role
     @vacols_role ||= self.class.user_repository.vacols_role(css_id)
   end
 
+  def vacols_attorney_id
+    @vacols_attorney_id ||= self.class.user_repository.vacols_attorney_id(css_id)
```
[the remark this answers is not in the document]
Comment (aroltsch) [reply app/models/user.rb]: Great suggestion!

File: app/models/attorney_case_reviews/attorney_case_review.rb (hunk 1 above)
Comment (aroltsch) [reply app/models/attorney_case_reviews/attorney_case_review.rb]: Yes, please! I tested it but I'd like a second pair of eyes.

File: app/repositories/queue_repository.rb [hunk 6]
```diff
@@ -34,6 +40,30 @@ def self.appeals_from_tasks(tasks)
...
+        deteam: attorney.vacols_group_id,
+        deadusr: judge.vacols_uniq_id,
+        deadtim: VacolsHelper.local_date_with_utc_timezone,
+        deicr: decass_complexity_rating(vacols_id)
+      )
+    end
+  end
+
+  def self.decass_complexity_rating(vacols_id)
+    VACOLS::Case.select("VACOLS.DECASS_COMPLEX(bfkey) as complexity_rating")
```
[the remark this answers is not in the document]
Comment (aroltsch) [reply app/repositories/queue_repository.rb]: it does!

File: app/models/attorney_case_reviews/attorney_case_review.rb (hunk 1 above)
Comment (amprokop) [reply app/models/attorney_case_reviews/attorney_case_review.rb]: 👍

File: app/repositories/queue_repository.rb [hunk 7]
```diff
@@ -34,6 +40,30 @@ def self.appeals_from_tasks(tasks)
...
+      vacols_case = VACOLS::Case.find(vacols_id)
+      vacols_case.update_vacols_location!(vacols_id)
+      vacols_case.update(bfattid: attorney.vacols_attorney_id)
+
+      VACOLS::Decass.create!(
+        defolder: vacols_id,
+        deatty: attorney.vacols_attorney_id,
+        deteam: attorney.vacols_group_id,
+        deadusr: judge.vacols_uniq_id,
+        deadtim: VacolsHelper.local_date_with_utc_timezone,
```
Comment (aroltsch) [inline app/repositories/queue_repository.rb]: I forgot to set `deassign` here!

**Comment by amprokop** [comment]
## Testing

### Steps to repro on dev `local` env
```
vacols_id = "3626186"
appeal_id = Appeal.find_by(vacols_id: vacols_id).id
attorney = User.find_by(css_id: "BVASRITCHIE")
judge = Judge.list_all[8]
RequestStore.store[:current_user] = judge

JudgeCaseAssignment.assign_to_attorney!(appeal_type: "Legacy", assigned_by: judge, assigned_to: attorney, appeal_id: appeal_id)
```

### Result
```
ActiveRecord::RecordInvalid: Validation failed: Deicr can't be blank
```

It looks like this will always fail on local FACOLS, since
`
VACOLS::Case.select("VACOLS.DECASS_COMPLEX(bfkey) as complexity_rating").map { |a| a.complexity_rating }.compact.length` returns 0. Do we need to find a way to import the complexity_rating function? Or stub it for now?

File: app/repositories/queue_repository.rb [hunk 8]
```diff
@@ -34,6 +40,32 @@ def self.appeals_from_tasks(tasks)
     appeals
   end
 
+  def self.assign_case_to_attorney!(judge:, attorney:, vacols_id:)
+    # TODO: add depdiff
+    transaction do
```
Comment (aroltsch) [inline app/repositories/queue_repository.rb]: I will add the metrics service in the datadog PR.

**Pull request merged**